scripts/mbedtls_framework/code_wrapper/psa_wrapper.py

```diff
@@ -0,0 +1,283 @@
+        'psa_key_derivation_verify_bytes', # not implemented yet
+        'psa_key_derivation_verify_key', # not implemented yet
+        'psa_set_key_domain_parameters', # client-side function
+    ])
+    # PAKE stuff: not implemented yet
+    not_implemented = frozenset([
+        'psa_crypto_driver_pake_inputs_t *',
+        'psa_pake_cipher_suite_t *',
+    ])
+
+    function_guards = {
+        'mbedtls_psa_register_se_key': 'defined(MBEDTLS_PSA_CRYPTO_SE_C)',
+        'mbedtls_psa_inject_entropy': 'defined(MBEDTLS_PSA_INJECT_ENTROPY)',
+        'mbedtls_psa_external_get_random': 'defined(MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG)',
+        'mbedtls_psa_platform_get_builtin_key': 'defined(MBEDTLS_PSA_CRYPTO_BUILTIN_KEYS)',
+        'psa_crypto_driver_pake_get_cipher_suite' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_password' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_password_len' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_peer' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_peer_len' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_user' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_user_len' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_abort' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_get_implicit_key' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_input' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_output' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_password_key' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_peer' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_role' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_user' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_setup' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+    }
+
+class PSAWrapper(c_wrapper_generator.Base):
+    """Generate a C source file containing wrapper functions for PSA Crypto API calls."""
+
+    _WRAPPER_NAME_PREFIX = 'mbedtls_test_wrap_'
+    _WRAPPER_NAME_SUFFIX = ''
+
+    _PSA_WRAPPER_INCLUDES = ['<psa/crypto.h>']
+
+    def __init__(self,
+                 out_h_f: str,
+                 out_c_f: str,
+                 in_headers: List[str] = PSAWrapperCFG.input_headers,
+                 config: PSAWrapperCFG = PSAWrapperCFG()) -> None:
+
+        super().__init__()
+        self.in_headers = in_headers
+        self.out_c_f = out_c_f
+        self.out_h_f = out_h_f
+
+        self.mbedtls_root = build_tree.guess_mbedtls_root()
+
+        self.read_config(config)
+
+        if in_headers and not config:
+            self.read_headers(in_headers)
+
+    def read_config(self, cfg: PSAWrapperCFG)-> None:
+        """Configure instance's parameters from a user provided config."""
+        if not cfg:
+            return
+        self._CPP_GUARDS = PSAWrapper.parse_def_guards(cfg.define_guards)
```

Comment:
Which is the private file here? neither `self._CPP_GUARDS` nor  `cfg.define_guards` fit that description?

Reply:
Here the point is _lowercase_ instead of uppercase. The name should be `self.cpp_guards` to convey that it's ok to access the field from outside the class, or `self._cpp_guards` to convey that only methods of the class or a descendant should access the field.

Reply:
This is still not addressed as of 03bfc58000f7e6ff35a0f4a97821619c054ee275.

Reply:
(continuing at https://github.com/Mbed-TLS/mbedtls-framework/pull/25#discussion_r1682437015)